### DISCUSSION TITLE
Application access on teleport

### DISCUSSION THREAD

**Discussion by sanchezzjf** [gravitational/teleport#32902] [Q&A] [answered]
I have a self-hosted community edition teleport instalation and I have configured it to access our web applications, but some of them are proxied(like grafana.example.com) and some of them it just redirects to the page requested.

How does that work? Why some applications just get redirected and not properly proxied?

**Comment by webvictim** [comment, marked as answer]
If your applications are sending redirect headers to the non-proxied address of the application, the browser will still follow those.

You might want to use a `rewrite.redirect` configuration like this:

```yaml
app_service:
  enabled: true
  apps:
  - name: internalapp
    uri: http://internal.company.com:8080
    public_addr: internalapp.teleport.example.com
    rewrite:
      redirect:
      - internal.company.com
      - otherapp.internal
```

This means that if the app ever sends a header to the browser saying "you need to go to http://internal.company.com" or "you need to go to http://otherapp.internal", Teleport will rewrite those messages to say "you need to go to https://internalapp.teleport.example.com" instead.

I hope this makes sense.

The other way to try and prevent this is to change the app's configuration to tell it that its own address is "internalapp.teleport.example.com" rather than "internal.company.com". This way it can send the correct redirects itself.

Note that there's no way to totally guarantee that the app can't send _any_ redirects at all to `internal.company.com` - Teleport tries to catch `Location` header rewrites, but there are ways to do this via Javascript and HTML which it can't intercept. Every app is different and they use lots of different methods.

The best way to make sure that users can't use an app without going through Teleport is to firewall off the app so that it only accepts incoming traffic from the IP of the host where the Teleport `app_service` is running. That way if users are redirected to the internal address of the app, they won't be able to connect to it.

**Reply by sanchezzjf** [reply]
Thanks, that was pretty clarifying.

**Comment by mind-drive** [comment]
Does a reverese proxied application can still be recorded like ssh?
If not, is it possible to record the web browser activity thorugh teleport?

**Reply by webvictim** [reply]
See details here of the audit logging available for application sessions: https://github.com/gravitational/teleport/discussions/11334#discussioncomment-2415029

TL;DR No, it's not possible to get a console or web browser recording of application sessions, but you can see all the HTTP requests made by the user.
